OutOfSchool/OutOfSchool.WebApi/Controllers/ChatWorkshopController.cs

```diff
@@ -0,0 +1,470 @@
+
+                    if (messages.Any())
+                    {
+                        return Ok(messages);
+                    }
+                    else
+                    {
+                        return NoContent();
+                    }
+                }
+                else
+                {
+                    return StatusCode(403, "Forbidden to get a chat room of another users.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get a list of ChatRooms for current user.
+        /// </summary>
+        /// <returns>List of ChatRooms with last message and number of not read messages.</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IEnumerable<ChatRoomWorkshopDtoWithLastMessage>))]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<IActionResult> GetUsersRoomsAsync()
+        {
+            var userId = HttpContext.User.FindFirst("sub")?.Value;
+            var userRole = HttpContext.User.FindFirst("role")?.Value;
+
+            if (userId is null || userRole is null)
+            {
+                throw new ArgumentException($"{nameof(userId)} or {nameof(userRole)}");
+            }
+
+            IEnumerable<ChatRoomWorkshopDtoWithLastMessage> chatRooms;
+
+            if (string.Equals(userRole, Role.Provider.ToString(), StringComparison.OrdinalIgnoreCase))
+            {
+                var providerId = await validationService.GetEntityIdAccordingToUserRoleAsync(userId, userRole).ConfigureAwait(false);
+                chatRooms = await roomService.GetByProviderIdAsync(providerId).ConfigureAwait(false);
+            }
+            else
+            {
+                var parentId = await validationService.GetEntityIdAccordingToUserRoleAsync(userId, userRole).ConfigureAwait(false);
+                chatRooms = await roomService.GetByParentIdAsync(parentId).ConfigureAwait(false);
+            }
+
+            if (!chatRooms.Any())
+            {
+                return NoContent();
+            }
+
+            return Ok(chatRooms);
+        }
+
+        /// <summary>
+        /// Set status IsRead on true for each user's ChatMessage in the ChatRoom.
+        /// </summary>
+        /// <param name="chatRoomId">ChatRoom's Id.</param>
+        /// <returns>Number of successfully updated items.</returns>
+        [HttpPut]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(int))]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<IActionResult> UpdateMessagesStatusAsync(long chatRoomId)
```

Comment:
why do we want to have such method ? 
If the main reason to update all when user opened a chat page - what if this page has a pagination, and user opened only the very first part of unread messages ?

Reply:
let's ask BA to concretize the requirements.
Added to TODO list